### ISSUE TITLE
비디오 센서 싱크

### ISSUE THREAD

**Issue by Doldolee** [sung-park/sensor_data_labeler#24]
json파싱 데이터로 하여도 싱크가 완벽히 정확하지는 않은 것 같습니다.
공유드리는 영상과 같이 반려견이 내려옴과 동시에 파형이 요동쳐야하는데 내려온 후에 파형이 요동치는 것을 알 수 있습니다.
꼬미와 사랑이 데이터를 살펴보았을 때 꼬미 데이터는 이러한 문제가 거의 없는 것으로 보이는데 사랑이 데이터는 좀 차이가 있는 것 같습니다.(기기마다 차이가 있는것으로 추정됩니다) 
https://youtu.be/gWGZGe57Aho?si=r43BPSoiK4OS7hDh

특이한 점은 사랑이의 경우 json상에서 센서가 늦게 켜지는 것으로 보여 비디오 시작 지점을 센서에 맞출경우 싱크 차이가 더 커진다는 것입니다. 실제로 offset을 맞추기 전과 offset을 맞춘 후의 데이터 2가지를 비교해보면 offset을 맞추기 전이 좀 더 싱크가 맞습니다.(그래서 저는 전처리할 때 센서가 늦게 켜지는 경우는 offset을 맞추지 않았습니다). 저는 사랑이와 꼬미 데이터만 비교하였기 때문에 이러한 경우가 일반적인 문제인지 파악해볼 필요는 있어보입니다.
그런데 이 문제들이 json데이터 기록 상의 문제여서 해당 문제를 클리어하게 해결하기는 어려울 것 같다는 생각이 들기는 합니다.

이러한 에러가 긴 구간을 라벨링할 때는 크게 문제가 되지 않지만 비교적 순간 동작을 라벨링할 때는 문제가 될 수도 있습니다. 그런데 해당 문제를 보완할 방법이 마땅치가 않아서 일단 라벨러가 순간 동작을 라벨링할 경우 어느 정도 차이(0.1~0.3초)가 있을 수 있다라는 점만 공유하고 시그널을 잘 보고 라벨링을 해야할 것 같습니다.

추가로 현재 json파일에서 센서 시작과 동영상 시작 시간을 파싱하고있는 것 같습니다. 그런데 json의 센서시작 시간과 csv파일의 시작 시간도 차이가 있기 때문에 csv파일의 센서 시작시간과 json의 동영상 시작 시간 데이터를 가지고 offset을 맞추는게 좀 더 정확해질 것 같습니다.

**Comment by sung-park** [comment]
> https://youtu.be/gWGZGe57Aho?si=r43BPSoiK4OS7hDh

혹시 파일명이 어떻게 될까요?

**Comment by Doldolee** [comment]
_[2023-10-11] 14_19_20_550.csv

해당 파일명 입니다

**Comment by Doldolee** [comment]
네 맞습니다. 해당 offset을 rule로 정의할 수 있을려면 다른 기기 데이터도 비교하며 판단해봐야할 것 같습니다

**Comment by sung-park** [comment]
네 현재 사랑이 시료가 배포 중이라 직접 확인이 어렵고 다른 시료를 좀 살펴봐야겠네요. 폰과 와치의 시스템 시간이 정확히 싱크되지 않아서 발생할 수도 있는 것 같은데 은근히 확인이 까다롭네요.

**Comment by sung-park** [comment]
동일 WiFi에 폰 2개와 와치 1개를 연결해도 아래와 같이 시스템 시간 차이가 있네요. 이걸 강제로 맞추는 방법 있는지 알아봐야할 것 같아요.

R3CN1051TNZ:	                1697796722.321 #폰2
ce0717175520743a0d7e:	1697796722.556 #폰1-와치
RFAT51PNDBN:	                1697796722.694 #와치

**Comment by Doldolee** [comment]
해당 문제 원인 파악이 되었을까요? 파악되었다면 이슈 닫도록 하겠습니다!
